### PR TITLE
bug/cast-ids-to-string

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- #26 
- #27 

**This PR will result in the following new package version:**

- v0.8.0 because of casting changes.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## Breaking Changes
> - Casted `*_id` fields as strings in upstream `stg_*` models to ensure compatibility in downstream joins.
> 
> ## Documentation
> - Update dbt documentation definitions.
> - Update README formatting.
> 
> ## Under the Hood
> - Added an additional integration test to verify functionality when `greenhouse_using_*` variables are disabled.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] ~dbt run (if incremental models are present) && dbt test~

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

- I was able to recreate the issue with the seed data in Big Query:
  - Screenshot 2024-11-20 at 4 21 11 PM

- The updates resolve the error, however I was unable to run consistency tests due to the cast changes.

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:1]: Reminder to update the `Does this package have dependencies?` section with the new version range for the source package.

File: packages.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 packages:
-  - package: fivetran/greenhouse_source
-    version: [">=0.7.0", "<0.8.0"]
+  # - package: fivetran/greenhouse_source
+  #   version: [">=0.8.0", "<0.9.0"]
+
+- git: https://github.com/fivetran/dbt_greenhouse_source.git
+  revision: bug/cast-ids-to-string
+  warn-unpinned: false
```
Comment (fivetran-joemarkiewicz) [inline packages.yml:7]: Reminder to swap before merge

File: .gitignore [hunk 2]
```diff
@@ -2,4 +2,5 @@ target/
 dbt_modules/
 logs/
 .DS_Store
-dbt_packages/
\ No newline at end of file
+dbt_packages/
+package-lock.yml
```
Comment (fivetran-joemarkiewicz) [inline .gitignore:6]: Request to also add the virtual environment to the gitignore.
```suggestion
package-lock.yml
env/
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,17 @@
+# dbt_greenhouse v0.8.0
+[PR #28](https://github.com/fivetran/dbt_greenhouse/pull/28) is a breaking change due to [upstream updates](
+https://github.com/fivetran/dbt_greenhouse_source/releases/tag/v0.8.0):
+
+## Breaking Changes
+- Casted `*_id` fields as strings in upstream `stg_*` models to ensure compatibility in downstream joins.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:6]: Similar note from the [source PR notes](https://github.com/fivetran/dbt_greenhouse_source/pull/31#discussion_r1852992296), can you callout that the majority of these ID fields were initially integers and that they should expect the downstream fields in the end models will now be strings as opposed to integers.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz thanks for working on this PR! Just a few small comments and requests before approval. Let me know once it's ready for re-review!

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-joemarkiewicz thanks--I have updated!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,18 @@
...
+## Breaking Changes
+- Updated `*_id` fields in upstream `stg_*` models to be cast as strings to ensure compatibility in downstream joins by avoiding potential type mismatches.
+  - Note: most IDs were previously stored as integers, so `*_id` fields in the end models will now also be strings.
+
+## Documentation
+- Update dbt documentation definitions.
+- Update README formatting.
+
+## Under the Hood
+- Added an additional integration test to verify functionality when `greenhouse_using_*` variables are disabled.
```
Comment (fivetran-avinash) [inline CHANGELOG.md:14]: See comment in the source PR.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,18 @@
+# dbt_greenhouse v0.8.0
+[PR #28](https://github.com/fivetran/dbt_greenhouse/pull/28) is a breaking change due to [upstream updates](
+https://github.com/fivetran/dbt_greenhouse_source/releases/tag/v0.8.0):
+
+## Breaking Changes
+- Updated `*_id` fields in upstream `stg_*` models to be cast as strings to ensure compatibility in downstream joins by avoiding potential type mismatches.
+  - Note: most IDs were previously stored as integers, so `*_id` fields in the end models will now also be strings.
+
+## Documentation
+- Update dbt documentation definitions.
```
Comment (fivetran-avinash) [inline CHANGELOG.md:10]: Is this referring to the yml updates? Looks like fields were added but no definitions were updated.

**Review by fivetran-avinash — APPROVED** [review]
@fivetran-catfritz A few comments but nothing blocking, lgtm

File: CHANGELOG.md (hunk 5 above)
Comment (fivetran-catfritz) [reply CHANGELOG.md:10]: `greenhouse.yml` was updated. Or is the issue that you want it to be "added" instead of "updated".

**Pull request merged**